tests/core/test_core.py

```diff
@@ -406,20 +406,26 @@ def test_copy_to_clipboard_exception(
         assert popup.call_args_list[0][0][1] == "area:error"
 
     @pytest.mark.parametrize(
-        "url",
+        "url, name, name_attribute_present",
         [
-            "https://chat.zulip.org/#narrow/stream/test",
-            "https://chat.zulip.org/user_uploads/sent/abcd/efg.png",
-            "https://github.com/",
+            ("https://chat.zulip.org/#narrow/stream/test", "mozilla", True),
+            ("https://chat.zulip.org/user_uploads/sent/abcd/efg.png", "chrome", True),
+            ("https://github.com/", "default browser", False),
         ],
     )
     def test_open_in_browser_success(
-        self, mocker: MockerFixture, controller: Controller, url: str
+        self,
+        mocker: MockerFixture,
+        controller: Controller,
+        url: str,
+        name: str,
+        name_attribute_present: bool,
     ) -> None:
         # Set DISPLAY environ to be able to run test in CI
         os.environ["DISPLAY"] = ":0"
         mocked_report_success = mocker.patch(MODULE + ".Controller.report_success")
         mock_get = mocker.patch(MODULE + ".webbrowser.get")
+        mock_get.return_value.name = name
```

Comment:
Changing it so that I'm not explicitly setting the `mock_get.return_value.name` at all and adding a condition to check if `expected_name` is false (since None is a falsy value) and setting the `mock_get.return_value.name` as "default browser".

This way we can illustrate the behaviour change and the existing test passes even without changing anything else.

Reply:
The point I was making is that we want the name attribute to be conditionally *absent*, representing the macos case for python <3.11.

The value we assign here is what `mock_get`, ie. `webbrowser.get()`, will have as a return value - so hopefully you can see that setting it to be `"default browser"` in the conditional doesn't represent the situation to test?